backend/firestore/firestore.go

```diff
@@ -0,0 +1,174 @@
+package firestore
+
+import (
+	"context"
+	"time"
+
+	fs "cloud.google.com/go/firestore"
+	"github.com/kellegous/go/internal"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// NextID is the next numeric ID to use for auto-generated IDs
+type NextID struct {
+	ID uint32 `json:"id" firestore:"id"`
+}
+
+// Backend provides access to Google Firestore.
+type Backend struct {
+	db *fs.Client
+}
+
+// New instantiates a new Backend
+func New(path string) (*Backend, error) {
+	if path == "" {
+		path = getGoogleProject()
```

Comment:
If you pass in the `Context`, you can also propagate it to this.

Reply:
The passed in context will be `context.Background()` with no timeout; I wasn't wanting that to be the case for this call.